packages/core/src/odata-common/request/odata-request.ts

```diff
@@ -55,7 +55,9 @@ export class ODataRequest<RequestConfigT extends ODataRequestConfig> {
    * @returns The absolute URL for the request
    */
   url(): string {
-    return `${removeTrailingSlashes(this.resourceUrl())}${this.query()}`;
+    return `${removeTrailingSlashes(this.resourceUrl())}${removeTrailingSlashes(
+      this.config.appendPaths.join('/')
```

Comment:
is ensured that the path which are joined here do not contain trailing slashes? Test would be good.

Reply:
what happens if user includes query parameters in path?